File: codechef_nov12/conflip.py
import sys

rl = sys.stdin.readline
t = int(rl())

for _ in range(t):
	g = int(rl())
	for _ in range(g):
		start, N, count = map(int, sys.stdin.readline().split())
		if N % 2 == 0:
			print N/2
		else:
			if start == count:
				print N/2
			else:
				print N/2 + 1

File: codechef_nov12/conflip.cpp
#include <iostream>

using namespace std;

int main() {
	int t, g, start, q;
	long n;
	cin >> t;
	for (int i = 0; i < t; ++i) {
		cin >> g;
		for (int j = 0; j < g; ++j) {
			cin >> start >> n >> q;
			if (start != q)
				cout << (n >> 1) + (n & 1) << endl;
			else
				cout << (n >> 1) << endl;
		}
	}
}